lib/optimizely/event/forwarding_event_processor.rb

```diff
@@ -30,11 +30,14 @@ def process(user_event)
       log_event = Optimizely::EventFactory.create_log_event(user_event, @logger)
 
       begin
-        @event_dispatcher.dispatch_event(log_event)
-        @notification_center&.send_notifications(
-          NotificationCenter::NOTIFICATION_TYPES[:LOG_EVENT],
-          log_event
-        )
+        @notification_center&.send_notifications(NotificationCenter::NOTIFICATION_TYPES[:LOG_EVENT], log_event)
+        Thread.new do
+          begin
+            @event_dispatcher.dispatch_event(log_event)
+          rescue StandardError => e
```

Comment:
Can we run into second rescue when we have rescued here?

Reply:
The first rescue is for the event dispatch failure.  really just for logging.  the second is for thread failure.